source/js/diva.js

```diff
@@ -943,45 +943,78 @@ module.exports = diva;
+	            var outerBottom = outerOffset.bottom;
+	            var outerRight = outerOffset.right;
+	
+	            //if the clicked position was outside the diva-outer object, it was not on a visible portion of a page
+	            if (pageX < outerLeft || pageX > outerRight)
+	                return -1;
+	
+	            if (pageY < outerTop || pageY > outerBottom)
+	                return -1;
+					
+				// Get the current page index
+				var currentPageIndex = this.getCurrentPageIndex();
+				// Get the current page dimension
+				
+				var pageDimension = this.getPageDimensionsAtCurrentZoomLevel(currentPageIndex);
+	            if (settings.inGrid) {
+					// Calculate the position of top left corner of the current page 
+					var yOffset = outerOffset.height / 2 - viewerState.verticalOffset + outerTop;
+					var xOffset = outerOffset.width / 2 - viewerState.horizontalOffset + outerLeft; 
+					
+					// Calculate the row and column index relative to current page index
+					var colIndex = Math.floor((pageX - xOffset) / pageDimension.width);
+					var rowIndex = Math.floor((pageY - yOffset) / pageDimension.height);
+					// Using the pages per row to calculate the page index
+					var clickPageIndex = rowIndex * settings.pagesPerRow + colIndex + currentPageIndex;
+					if (clickPageIndex > viewerState.numPages) {
+						return -1;
+					}
+					return clickPageIndex;
+				} else if (settings.inBookLayout) {
+					// Calculate the position of top left corner of the current page 
+					var yOffset = outerOffset.height / 2 - viewerState.verticalOffset + outerTop;
+					var xOffset = outerOffset.width / 2 - viewerState.horizontalOffset + outerLeft; 
+					
+					// Calculate the row and column index relative to current page index
+					var colIndex = Math.floor((pageX - xOffset) / pageDimension.width);
+					var rowIndex = Math.floor((pageY - yOffset) / pageDimension.height);
+					// Using the pages per row to calculate the page index
+					var clickPageIndex = rowIndex * 2 + colIndex + currentPageIndex;
+					if (clickPageIndex > viewerState.numPages) {
+						return -1;
+					}
+					return clickPageIndex;
+				} else {
+
+					// Calculate the position of top left corner of the current page 
+					var yOffset = outerOffset.height / 2 - viewerState.verticalOffset + outerTop;
+					var xOffset = outerOffset.width / 2 - pageDimension.width / 2 + outerLeft; 
+					
+					if (pageX < xOffset || pageX > xOffset + pageDimension.width) {
+						return -1;
+					}
+					
+					// Calculate the row and column index relative to current page index
+					var colIndex = Math.floor((pageX - xOffset) / pageDimension.width);
+					var rowIndex = Math.floor((pageY - yOffset) / pageDimension.height);
+					// Using the pages per row to calculate the page index
+					var clickPageIndex = rowIndex + currentPageIndex;
+					if (clickPageIndex > viewerState.numPages) {
+						return -1;
+					}
+					return clickPageIndex;
+		
+				}
+
+	            return -1;
```

Comment:
you already have:
`if (...) and then else { return something }` so the return on line 1016 will never be reached.  You can remove the line.